### PR TITLE
CFE-3315: Added execresult_as_data()

### PR DESCRIPTION
**Pull request by olehermanse**
Merge together:
https://github.com/cfengine/core/pull/4268
https://github.com/cfengine/nova/pull/1695
https://github.com/cfengine/documentation/pull/2344

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5452/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5452/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5452/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5453/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5453/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5453/

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5454/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5454/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5454/

File: libpromises/evalfunction.c [hunk 1]
```diff
@@ -2359,12 +2362,26 @@ static FnCallResult FnCallExecResult(ARG_UNUSED EvalContext *ctx, ARG_UNUSED con
...
+            free(buffer);
+            return res;
+        }
+        else
+        {
+            assert(StringEqual(function, "execresult_as_data"));
+            JsonElement *result = JsonObjectCreate(2);
+            JsonObjectAppendInteger(result, "exit_code", exit_code);
+            JsonObjectAppendString(result, "output", buffer);
+            free(buffer);
```
Comment (vpodzime) [inline libpromises/evalfunction.c:2382]: Do we have a function to append a string without creating a copy?

Comment (olehermanse) [reply libpromises/evalfunction.c:2382]: Nope

**Pull request merged**